test/realm_test.dart

```diff
@@ -1551,6 +1550,210 @@ Future<void> main([List<String>? args]) async {
+    final originalRealm = getRealm(originalConfig);
+
+    final app = App(appConfiguration);
+    final credentials = Credentials.anonymous(reuseCredentials: false);
+    var user = await app.logIn(credentials);
+    final configCopy = Configuration.flexibleSync(user, [Product.schema]);
+    expect(() => originalRealm.writeCopy(configCopy),
+        throws<RealmException>("Realm cannot be converted to a flexible sync realm unless flexible sync is already enabled"));
+  });
+
+  test('Realm writeCopy Local->Local inside a write block', () {
+    final originalConfig = Configuration.local([Car.schema]);
+    final originalRealm = getRealm(originalConfig);
+    final pathCopy = originalConfig.path.replaceFirst(p.basenameWithoutExtension(originalConfig.path), generateRandomString(10));
+    final configCopy = Configuration.local([Car.schema], path: pathCopy);
+    var itemsCount = 10;
+    originalRealm.write(() {
+      for (var i = 0; i < itemsCount; i++) {
+        originalRealm.add(Car("make_${i + 1}"));
+      }
+      originalRealm.writeCopy(configCopy);
+    });
+    originalRealm.close();
+
+    expect(File(pathCopy).existsSync(), isTrue);
+    final copiedRealm = getRealm(configCopy);
+    expect(copiedRealm.all<Car>().length, itemsCount);
+    expect(copiedRealm.isClosed, isFalse);
+    copiedRealm.close();
+  });
+
+  test('Realm writeCopy Local->Local read-only', () {
+    final originalConfig = Configuration.local([Car.schema]);
+    final originalRealm = getRealm(originalConfig);
+    final pathCopy = originalConfig.path.replaceFirst(p.basenameWithoutExtension(originalConfig.path), generateRandomString(10));
+    final configCopy = Configuration.local([Car.schema], path: pathCopy, isReadOnly: true);
+
+    originalRealm.writeCopy(configCopy);
+    originalRealm.close();
+
+    expect(File(pathCopy).existsSync(), isTrue);
+    final copiedRealm = getRealm(configCopy);
+    expect(copiedRealm.isClosed, isFalse);
+    _expectAllWritesToThrow<RealmException>(copiedRealm, "Can't perform transactions on read-only Realms.");
+    _expectAllAsyncWritesToThrow<RealmException>(copiedRealm, "Can't perform transactions on read-only Realms.");
+
+    copiedRealm.close();
+  });
+
+  test('Realm writeCopy in-memory Local->Local', () {
+    final originalConfig = Configuration.inMemory([Car.schema]);
+    final originalRealm = getRealm(originalConfig);
+    final pathCopy = originalConfig.path.replaceFirst(p.basenameWithoutExtension(originalConfig.path), generateRandomString(10));
+    final configCopy = Configuration.local([Car.schema], path: pathCopy);
+
+    originalRealm.writeCopy(configCopy);
+    originalRealm.close();
+
+    expect(File(pathCopy).existsSync(), isTrue);
+    final copiedRealm = getRealm(configCopy);
+    expect(copiedRealm.isClosed, isFalse);
+    final itemsCount = 100;
+    copiedRealm.write(() {
+      for (var i = 0; i < itemsCount; i++) {
+        copiedRealm.add(Car("make_${i + 1}"));
```

Comment:
I am not sure why we are adding new data to the copied realm

Reply:
shouldn't we be testing that the new realm has the copied data?

Reply:
This proves that the new realm is in memory. I add new data then close it and I expect the newly added data to disapper.